### DISCUSSION TITLE
Feature: kill switch, version warnings/messages

### DISCUSSION THREAD

**Discussion by kyranjamie** [blockstack/stacks-wallet-web#1411] [Ideas]
Relevant for both desktop and web wallets.

## Problem
Once a release has been cut, we've no channel of communication to a user of that version. This can be problematic. We've found in the past that it would be hugely helpful if we had a way of toggling a message in the UI. Further, in the case of a security vulnerability, we want a way to quickly force users to upgrade to the latest patched version.

## Proposed solution
Clients could fetch JSON from a manually-maintained source, describing various messages or array of versions to kill.

## Implementation
We could keep this JSON file in our repo, and fetch it using the Github API. Modifications to the files are thus subject to the same rules/checks as our code changes.

**Demo json file api call**
`http https://raw.githubusercontent.com/blockstack/stacks-wallet-web/main/tsconfig.json`

The object could look something like this. Keyed by version number (or range):

```json
{
  "1.0.0": {
    "messages": [
      {
        "text": "Something terrible happened to version 1.0.0",
        "severity": "high",
        "dismissible": false
      }
    ],
    "killed": true
  }
}
```

We could also create a JSON Schema, so that the file is checked at both CI-time and runtime validating the input conforms to the schema.

**Comment by beguene** [comment]
I think it's a good solution. This will be helpful for many other features. 
I think we should make sure though that the wallet behave normally if this JSON endpoint does not respond.

**Comment by markmhendrickson** [comment]
This is a great idea, especially by combining the "end of life" switch we've needed in the past for an entire product line (e.g. v3 of the wallet) with the need to deprecate specific, more minor versions of the same product line (e.g. recent Ledger app compatibility upgrades). 

I also agree with @beguene that we ought to behave normally if the endpoint doesn't respond or responds with invalid data.
